katacomb/katacomb/util/image_util.py

```diff
@@ -1,215 +1,94 @@
-
-    # Minimum number of pixels allowed
-    minpix = max(min_npix, int(npix * max_reject))
-
-    # Grow rejected pixels in each iteration
-    ngrow = max(1, int(npix * 0.01))
-
-    if npix <= minpix:
-        return zmin, zmax
-
-    # Re-map indices from -1.0 to 1.0
-    # to improve fitting.
-    xscale = 2.0 / (npix - 1)
-    xnorm = np.arange(npix)
-    xnorm = xnorm * xscale - 1.0
-
-    # Set up for iteration
-    ngoodpix = npix
-    last_ngoodpix = npix + 1
-
-    # Mask used in k-sigma clipping.
-    badpix = np.zeros(npix, dtype=np.bool)
-
-    # Iterate, until maximum iteration, too many pixels
-    # rejected or until no change in number of good pixels
-    niter = 0
-    while niter < max_iter and ngoodpix > minpix and ngoodpix < last_ngoodpix:
-        # Fit a line to the remaining pixels
-        good_xnorm = xnorm[~badpix]
-        A = np.vstack((good_xnorm, np.ones(len(good_xnorm)))).T
-        slope, intercept = np.linalg.lstsq(A, samples[~badpix], rcond=None)[0]
-
-        # Subtract fitted line from the full data array
-        fitted = xnorm * slope + intercept
-        flat = samples - fitted
-
-        # Compute the k-sigma rejection threshold
-        sigma = np.std(flat[~badpix])
-        threshold = sigma * sigma_rej
-
-        # Detect and reject pixels further than k*sigma from the fitted line
-        badpix = np.abs(flat) > threshold
-
-        # Compute number of remaining pixels before convolution
-        last_ngoodpix = ngoodpix
-        ngoodpix = np.sum(~badpix)
-
-        # Convolve with a kernel of length ngrow
-        kernel = np.ones(ngrow, dtype=np.bool)
-        badpix = np.convolve(badpix, kernel, mode='same')
-
-        niter += 1
-
-    # Transform the slope back to the X range [0:npix-1]
-    slope = slope * xscale
-
-    # Apply contrast scaling
-    if contrast > 0:
-        slope = slope / contrast
-
-    # Stretch the slope
-    slope_hi = slope * stretch
-    slope_low = slope / stretch
-
-    # Median of the remaining samples is close to true
-    # pixel median sans sources/artefacts
-    good_samples = samples[~badpix]
-    median = np.median(good_samples)
-
-    # Find the indices of the median, low and high pixels
-    median_pixel = np.searchsorted(samples, median)
-    low_pixel = np.searchsorted(samples, good_samples[0])
-    hi_pixel = np.searchsorted(samples, good_samples[-1])
-
-    # Derive scale limits from slope_low and slope_hi
-    z1 = max(zmin, median - (median_pixel - low_pixel) * slope_low)
-    z2 = min(zmax, median + (hi_pixel - median_pixel) * slope_hi)
-
-    return z1, z2
-
-
-def save_image(image, filename, plane=1, drop_edge=0.125, cmap='afmhot',
-               display_size=10., dpi=1000, **kwargs):
-    """
-    Write an image plane to a file using imshow with contrast scaling.
-    Input kwargs are passed to the zscale function for the contrast
-    scaling.
+use('Agg', warn=False)  # noqa: E402
+import matplotlib.axes
+from matplotlib import pylab as plt
```

Comment:
pylab [isn't recommended](https://matplotlib.org/api/pyplot_api.html) for programmatic use.
```suggestion
import matplotlib.pyplot as plt
```